vlm/src/test/scala/geotrellis/contrib/vlm/avro/GeotrellisRasterSourceSpec.scala

```diff
@@ -118,11 +117,10 @@ class GeotrellisRasterSourceSpec extends FunSpec with RasterMatchers with Better
     it("should have resolutions only for given layer name") {
       assert(
         sourceMultiband.resolutions.length ===
-          CollectionLayerReader(uriMultibandNoParams).attributeStore.layerIds.filter(_.name == layerId.name).length
-      )
-      assert(
-        new GeotrellisRasterSource(s"$uriMultibandNoParams?layer=bogusLayer&zoom=0").resolutions.length === 0
+          CollectionLayerReader(uriMultibandNoParams).attributeStore.layerIds.count(_.name == layerId.name)
       )
+
+      intercept[AttributeNotFoundError](new GeotrellisRasterSource(s"$uriMultibandNoParams?layer=bogusLayer&zoom=0").resolutions)
```

Comment:
I want also to admit that this behavior change caused because all the metadata will be fetched at once now.